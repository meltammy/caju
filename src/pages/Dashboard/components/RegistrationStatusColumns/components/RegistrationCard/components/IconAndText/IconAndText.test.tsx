import "jest-styled-components";
import "@testing-library/jest-dom";
import { render, screen } from "@testing-library/react";
import { IconAndText } from ".";

const MockIcon = () => <svg data-testid="mock-icon" />;
const testText = "Sample Text";

describe("IconAndText", () => {
  it("renders the icon and text correctly", () => {
    render(<IconAndText icon={<MockIcon />} text={testText} />);

    expect(screen.getByTestId("mock-icon")).toBeInTheDocument();
    expect(screen.getByText(testText)).toBeInTheDocument();
  });

  it("should match snapshot when loading", () => {
    const { asFragment } = render(
      <IconAndText icon={<MockIcon />} text={testText} />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
