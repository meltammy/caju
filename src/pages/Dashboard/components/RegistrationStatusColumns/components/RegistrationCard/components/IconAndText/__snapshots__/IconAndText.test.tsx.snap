// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconAndText should match snapshot when loading 1`] = `
<DocumentFragment>
  .c0 {
  display: flex;
  align-items: center;
  gap: 0.5rem;
  padding: 0 1rem;
  font-size: 1rem;
}

.c0 svg {
  min-width: 1rem;
}

.c0 span {
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
}

.c0:first-of-type {
  margin-bottom: 0.5rem;
  padding: 1rem 3rem 0.5rem 1rem;
  border-bottom: 1px black solid;
  font-size: 2rem;
}

.c0:first-of-type span {
  margin: 0;
  font-size: 1.5rem;
  font-weight: 600;
  line-height: 1.5rem;
}

<div
    class="c0"
  >
    <svg
      data-testid="mock-icon"
    />
    <span>
      Sample Text
    </span>
  </div>
</DocumentFragment>
`;
